### ISSUE TITLE
Downloading files from a non-supported protocol URL (e.g. svn)

### ISSUE THREAD

**Issue by magicmyth** [flatpak/flatpak-builder#133]
I've made a flatpak-builder manifest for the GtkRadiant (a ID Tech game engine level editor) app here:
https://github.com/magicmyth/io.github.TTimo.GtkRadiant
Its an early first attempt at it so plenty still to do.

Currently its build process downloads game paks via svn (svn gets built by the manifest as well). I know downloading during the actual build process is not recommended but at the moment that's just how this project works but I do intend to change that part. The problem is these game paks are only distributed over svn or git and must be present during build for GtkRadiant to support the game. I would like to move the grabbing of the game paks to a separate source or module part of the phase and the git based ones should be easy to do as there is support for git URLs built in but as flatpak-builder does not seem to do svn I can't currently do that. I could do a shell type source with the _build-arg --share=network_ and run the svn checkout command that way but would that really be any different to adding network access to the apps build process as I am currently doing?

So I was wondering if it is possible to expose other kinds of file retrieval protocols to builder? Is it just recommended that such scenarios use shell or script types with network access to grab such sources? If so how does one go about exposing that in a way the cache system would continue to function? I imagine there will be other scenarios where developers do not publish archives over http.

**Comment by TingPing** [comment]
>  Is it just recommended that such scenarios use shell or script types with network access to grab such sources?

No. The ideal solution is either `flatpak-builder` understands svn or the files get hosted as an archive somewhere else upstream.

**Comment by magicmyth** [comment]
That's what I expected (and agree). It would be useful if we could extend what kind of file retrieval protocols flatpak-builder can use though (git may not always be king a proprietary shops can often use strange VCs). However having support for svn would greatly expand the existing projects that can be easily supported.

**Comment by alexlarsson** [comment]
Adding a minimal svn source implementation should not be all that hard. Lemme give it a go today.

**Comment by magicmyth** [comment]
Thanks much appreciated. Though now I wish I had asked sooner so I would have saved time figuring out so many parts. Nah learnt a lot from that.

So am I correct in understanding if apps themselves used svn they would still need to either build svn into their flatpak or have it as a runtime extension? If the later I may have a go at making a runtime extension for the FDO runtime that apps can use. Or is that what a base app would be used for?

You are a super star for doing this.

**Comment by alexlarsson** [comment]
Yeah, svn is not part of the current sdk or runtime, so you have to include your own copy.

**Comment by alexlarsson** [comment]
Oh, sorry, missed one you don't want a runtime extension. Thats only for very special optional cases.
